cli/src/commands/extensions/api.rs

```diff
@@ -356,18 +361,21 @@ async fn create_project(
     #[string] name: String,
     #[string] group: Option<String>,
     #[string] repository_url: Option<String>,
+    #[string] organization: Option<String>,
 ) -> Result<CreatedProject> {
     let state = ExtensionState::from(op_state);
     let api = state.api().await?;
 
     // Retrieve the id if the project already exists, otherwise return the id or the
     // error.
-    match api.create_project(&name, group.clone(), repository_url).await {
-        Err(PhylumApiError::Response(ResponseError { code: StatusCode::CONFLICT, .. })) => api
-            .get_project_id(&name, group.as_deref())
-            .await
-            .map(|id| CreatedProject { id, status: CreatedProjectStatus::Exists })
-            .map_err(|e| e.into()),
+    match api.create_project(&name, organization.as_deref(), group.clone(), repository_url).await {
+        Err(PhylumApiError::Response(ResponseError { code: StatusCode::CONFLICT, .. })) => {
+            let org_group = Group::try_new(organization, group);
+            api.get_project_id(&name, org_group)
```

Comment:
It seems weird here that `create_project` takes organization and group strings, but `get_project_id` takes a `Group`. They should probably all take `&Group`?

Reply:
There's no necessity for `Group` in the `create_project` endpoint, so for the sake of simplicity it was avoided. While it would be fine to use `Group` here, I don't think there's any benefit to changing it other than introducing additional code on the callsite. So that would effectively just be a net negative?

Reply:
In the code for an extension to create a group, it takes the organization and group as separate strings and passes them to one function as separate strings and another function as a `Group`.

In general, I think it's better for two related values like this to be the same parameter so there are less parameters.

There are several places that duplicate the (fairly trivial) `combined_format` code when working with the organization and group as separate parameters.

Reply:
The combined format part is a valid argument, but this is something that will go away in the future so I don't want to design API boundaries around it.

Less parameters is good when it makes calling something easier. But in this scenario this just means having to create a new group in every place where we'd otherwise just pass the variables in directly otherwise. I'm not sure what we would gain here? At that point I'd rather remove the struct entirely, but that comes with the drawbacks of reduced API guarantees, since it was introduced **after** discovering a bug with the implementation when using separate parameters.

Reply:
The APIs being called are different and as such the implication of calling them incorrectly are also different. The places that don't use the `Group` type simply don't have the same issues as the places that do use it.

Reply:
`Group::try_new(org, group)` is a trivial amount of code that packages two pointers into a single value. The change to the caller is insignificant, all callees have one less positional parameter, callees can use functions implemented for `Group`, callees making API calls can match on an enum instead of a pair of options, and we can't accidentally transpose parameters except for the call to `try_new` which should always be consistent.

I didn't count, but it seems like there are more places where the current code is converting back and forth between `Group` and org+group than there are places that need to create a `Group` from external input.

This commit shows how the code is simplified by using `Group` consistently for functions that deal with specific groups: https://github.com/phylum-dev/cli/commit/b0699ed2378f179fe7f91acf1e4040b067fdba6a

I fixed a bug where the current `handle_update_project` is passing a `None` `org` to `api.update_project`, but a potentially `Some` `org` to all other functions it calls. It was obvious once the code was changed to use `Group`.

Reply:
> This commit shows how the code is simplified by using Group consistently for functions that deal with specific groups: https://github.com/phylum-dev/cli/commit/b0699ed2378f179fe7f91acf1e4040b067fdba6a

It really doesn't. Inlining code that never should be inlined, just goes to show that it clearly causes additional friction:

```
let group = Group::try_new(config.org(), matches.get_one::<String>("group"));
```

The main "benefit" you're gaining here is in reusing the functions for the combined path, which are irrelevant since they're not going to be there long-term anyway.